src/connectors/range/connectRange.ts

```diff
@@ -14,68 +22,128 @@ const withUsage = createDocumentationMessageGenerator(
 
 const $$type = 'ais.range';
 
-/**
- * @typedef {Object} CustomRangeWidgetOptions
- * @property {string} attribute Name of the attribute for faceting.
- * @property {number} [min = undefined] Minimal range value, default to automatically computed from the result set.
- * @property {number} [max = undefined] Maximal range value, default to automatically computed from the result set.
- * @property {number} [precision = 2] Number of digits after decimal point to use.
- */
-
-/**
- * @typedef {Object} RangeRenderingOptions
- * @property {function(Array<number, number>)} refine Sets a range to filter the results on. Both values
- * are optional, and will default to the higher and lower bounds. You can use `undefined` to remove a
- * previously set bound or to set an infinite bound.
- * @property {{min: number, max: number}} range Results bounds without the current range filter.
- * @property {Array<number, number>} start Current numeric bounds of the search.
- * @property {{from: function, to: function}} formatter Transform for the rendering `from` and/or `to` values.
- * Both functions take a `number` as input and should output a `string`.
- * @property {Object} widgetParams All original `CustomRangeWidgetOptions` forwarded to the `renderFn`.
- */
+export type Min = number | undefined;
+export type Max = number | undefined;
+export type Start = [Min, Max];
```

Comment:
Would `RangeBoundaries` make more sense?

```suggestion
export type RangeBoundaries = [Min, Max];
```

Reply:
I prefer boundaries, but the property in the renderState which is called `start`, so I went with the same

Reply:
It's not only used for `start` so it's fine for the name to differ: https://github.com/algolia/instantsearch.js/blob/0844f036cc51938a2ac6b3fccd6ff5ee948dc93a/src/connectors/range/connectRange.ts#L40